.github/workflows/flutter-ci.yml

```diff
@@ -0,0 +1,27 @@
+name: Flutter CI
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v2
+        with:
+          distribution: 'zulu'
+          java-version: '11'
+      - uses: subosito/flutter-action@v1
+      - name: Get dependencies
+        run: flutter pub get
+      - name: Check code format
+        run: flutter format --set-exit-if-changed .
+      - name: Run linter
+        run: flutter analyze .
+      - name: Run tests
+        run: flutter test
+      - name: Generate android APK
+        run: flutter build apk
```

Comment:
maybe removing this and adding it as another action to be executed after this one?

Reply:
It could be an option, but why are yo thinking about that? The idea of this action is check that the linters run and the app compiles, that is why I think the compile task is ok here. On the other hand, If we add another action, we'll have around the double of time to know that the PR compiles and is valid (because you have to do the flutter setup twice)

Reply:
I'll merge it now, if we want to split in a new job I will do in a new pr